spec/models/user_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { User.create(name: 'Donald', posts_counter: 0) }
+  let(:post) { user.posts.create(title: 'Post', comments_counter: 0, likes_counter: 0) }
+  it 'is valid with valid attributes' do
+    expect(user).to be_valid
+  end
+
+  it 'is invalid without a name' do
+    user.name = nil
+    expect(user).to_not be_valid
+  end
+
+  it 'is invalid with a negative posts_counter' do
+    user.posts_counter = -1
+    expect(user).to_not be_valid
+  end
+
+  it 'is valid with a zero posts_counter' do
+    user.posts_counter = 0
+    expect(user).to be_valid
+  end
```

Comment:
- [ ] @quavo19 Great work adding tests to user model validation, however, you forgot to add a test for the ` recent_posts` method. This is a project requirement as highlighted below:

![image](https://user-images.githubusercontent.com/61408860/236535756-9e05835d-6640-4ca8-a2e3-d94b99f75a83.png)

Reply:
hello, @danmainah those methods are private methods so they can't be accessed outside the function. that's why they can't be tested.
i can go ahead and remove `Private` from my files and submit if that is what you would like me to do